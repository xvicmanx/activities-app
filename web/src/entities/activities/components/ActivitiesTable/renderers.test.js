import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {
  DescriptionRenderer,
  CommunitiesSelectRenderer,
} from './renderers';

describe('ActivitiesTable renderers', () => {
  describe('DescriptionRenderer', () => {
    it('renders properly', () => {
      const renderer = new ShallowRenderer();
      const result = renderer.render(<DescriptionRenderer
        field={{
          name: 'test-name',
          value: 'test-value',
          onChange: jest.fn(),
        }}
      />);
      expect(result).toMatchSnapshot();
    });
  });

  describe('CommunitiesSelectRenderer', () => {
    it('renders properly', () => {
      const renderer = new ShallowRenderer();
      const result = renderer.render(<CommunitiesSelectRenderer
        field={{
          name: 'test-name',
          value: 'test-value',
          onChange: jest.fn(),
        }}
      />);
      expect(result).toMatchSnapshot();
    });
  });
});
