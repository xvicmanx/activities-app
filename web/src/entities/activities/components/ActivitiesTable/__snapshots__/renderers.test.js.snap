// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivitiesTable renderers CommunitiesSelectRenderer renders properly 1`] = `
<CommunitiesDropdown
  name="test-name"
  onChange={[MockFunction]}
  value="test-value"
/>
`;

exports[`ActivitiesTable renderers DescriptionRenderer renders properly 1`] = `
<textarea
  name="test-name"
  onChange={[MockFunction]}
  value="test-value"
/>
`;
